src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1452,4 +1460,76 @@ export class NetworkAreaDiagramViewer {
+                const edge: EdgeMetadata | undefined = this.diagramMetadata?.edges.find(
+                    (edge) => edge.equipmentId == branchState.branchId
+                );
+                if (edge === undefined) {
+                    console.warn('Skipping updating branch ' + branchState.branchId + ' labels: branch not found');
+                    return;
+                }
+                this.edgesMap.set(branchState.branchId, edge);
+            }
+            const edgeId = this.edgesMap.get(branchState.branchId)?.svgId ?? '-1';
+            this.setBranchSideLabel(branchState.branchId, '1', edgeId, branchState.value1);
+            this.setBranchSideLabel(branchState.branchId, '2', edgeId, branchState.value2);
+            this.setBranchSideConnection(branchState.branchId, '1', edgeId, branchState.connected1);
+            this.setBranchSideConnection(branchState.branchId, '2', edgeId, branchState.connected2);
+        });
+    }
+
+    private setBranchSideLabel(branchId: string, side: string, edgeId: string, value: number | string) {
+        const arrowGElement: SVGGraphicsElement | null = this.container.querySelector(
+            "[id='" + edgeId + '.' + side + "'] .nad-edge-infos g"
+        );
+        if (arrowGElement !== null) {
+            arrowGElement.classList.remove('nad-state-in', 'nad-state-out');
+            if (typeof value === 'number') {
+                arrowGElement.classList.add(DiagramUtils.getArrowClass(value));
+            }
+            const branchLabelElement = arrowGElement.querySelector('text');
+            if (branchLabelElement !== null && branchLabelElement !== undefined) {
+                branchLabelElement.innerHTML =
+                    typeof value === 'number' ? value.toFixed(this.getValuePrecision()) : value;
+            } else {
+                console.warn('Skipping updating branch ' + branchId + ' side ' + side + ' label: text not found');
+            }
+        } else {
+            console.warn('Skipping updating branch ' + branchId + ' side ' + side + ' label: label not found');
+        }
+    }
+
+    private getValuePrecision() {
+        const edgeInfoDisplayed = this.svgParameters.getEdgeInfoDisplayed();
+        switch (edgeInfoDisplayed) {
+            case EdgeInfoEnum.ACTIVE_POWER:
+            case EdgeInfoEnum.REACTIVE_POWER:
+                return this.svgParameters.getPowerValuePrecision();
+            case EdgeInfoEnum.CURRENT:
+                return this.svgParameters.getCurrentValuePrecision();
+            default:
+                return 0;
+        }
+    }
+
+    private setBranchSideConnection(branchId: string, side: string, edgeId: string, connected: boolean) {
+        const halfEdge: SVGGraphicsElement | null = this.container.querySelector("[id='" + edgeId + '.' + side + "']");
+        if (halfEdge !== null && halfEdge != undefined) {
+            if (connected == undefined || connected) {
```

Comment:
You can remove the undefined test as types 'boolean' and 'undefined' have no overlap

Reply:
The connected parameter actually can be undefined, added in the parameter definition